infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -23,6 +22,9 @@
     private LogReplicationConfig config;
     private CorfuRuntime runtime;
     private String remoteClusterId;
+    // Log tail when the current snapshot sync started.  We do not need to synchronize access to it because it will not
+    // be read(calculateRemainingEntriesToSend) and written(setBaseSnapshot) concurrently.
```

Comment:
it could be read and written concurrently, right? cause the calculation is scheduled and it could happen to coincide?